src/utils_request.js

```diff
@@ -0,0 +1,264 @@
+ *
+ * @param options.url
+ *  URL of the target endpoint. Supports both HTTP and HTTPS schemes.
+ * @param [options.method=GET]
+ *  HTTP method.
+ * @param [options.headers={}]
+ *  HTTP headers.
+ *  Note that the function generates several headers itself, unless
+ *  they are defined in the `headers` parameter, in which case the function leaves them untouched.
+ *  For example, even if you define `{ 'Content-Length': null }`, the function doesn't define
+ *  the 'Content-Length' header and the request will not contain it (due to the `null` value).
+  * @param [options.body]
+ *  HTTP payload for PATCH, POST and PUT requests. Must be a `Buffer` or `String`.
+ * @param [options.followRedirect=true]
+ *  Follow HTTP 3xx responses as redirects (default: true).
+ *  OPTIONALLY: This property can also be implemented as function which gets response object as
+ *  a single argument and should return `true` if redirects should continue or `false` otherwise.
+ * @param [options.maxRedirects=10]
+ *  The maximum number of redirects to follow.
+ * @param [options.removeRefererHeader=false]
+ *  Removes the referer header when a redirect happens.
+ *  If `true`, referer header set in the initial request is preserved during redirect chain.
+ * @param [options.encoding]
+ *  Encoding to be used on `setEncoding` of response data.
+ *  If `null`, the body is returned as a `Buffer`.
+ *  Anything else (including the default value of undefined) will be passed as the encoding parameter to `toString()`,
+ *  (meaning this is effectively utf8 by default).
+ *  (Note: if you expect binary data, you should set encoding: null.)
+ * @param [options.gzip=false]
+ *  If `true`, the function adds an `Accept-Encoding: gzip` header to request compressed content encodings from the server
+ *  (if not already present) and decode supported content encodings in the response.
+ *  Note that you can achieve the same effect by adding the `Accept-Encoding: gzip` header directly to `options.headers`,
+ *  similarly as `deflate` as `br` encodings.
+ * @param [options.json=false]
+ *  Sets body to JSON representation of value and adds `Content-type: application/json` header.
+ *  Additionally, parses the response body as JSON, i.e. the `body` property of the returned object
+ *  is the result of `JSON.parse()`. Throws an error if response cannot be parsed as JSON.
+ * @param [options.timeout]
+ *  Integer containing the number of milliseconds to wait for a server to send
+ *  response headers (and start the response body) before aborting the request.
+ *  Note that if the underlying TCP connection cannot be established, the OS-wide
+ *  TCP connection timeout will overrule the timeout option (the default in Linux can be anywhere from 20-120 seconds).
+ * @param [options.proxy]
+ *  An HTTP proxy to be used. Supports proxy authentication with Basic Auth.
+ * @param [options.strictSSL=true]
+ *  If `true`, requires SSL/TLS certificates to be valid.
+ * @param [options.abort]
```

Comment:
You didn't implement this option. Actually, `CheerioCrawler` is aborting requests that don't have HTML content-type **before** the data is downloaded, to avoid unnecessary overheads. The point of the `abort()` function was to provide support for this functionality.

Reply:
In order to do so. I would have to implement it in the requestBetter and check for the HTML content type here. I tried to make the requestBetter more generic to support multiple contentType. Should I add something like desiredContentType option, or just make this function to support only the html ?

Reply:
Actually I was thinking that having `abort()` function is the most generic solution - the user can decide whether to abort the request based on content-type or any other header or info.

Reply:
Ah, I think I got it now. the abort function is called always when the user passes it as an option and accepts the request as a parameter, so users decide what to do with the request. Am I right?

Reply:
Yeah exactly, maybe we can call it `abortFunction` to make it more clear.